### PR TITLE
Discriminator fixes

### PR DESCRIPTION
**Pull request by mcardosos**
Tiny little fixes.
- Some interface implementations weren't generating all needed funcs to satisfy the interface. ith this fix some cognitive services dataplanes can be generated.
- There was an interesting intersection with unmarshalling flattened models. This fixes the service fabric generation.
PTAL
@jhendrixMSFT @marstr @vladbarosan

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: src/Model/CompositeTypeGo.cs [hunk 1]
```diff
@@ -250,7 +250,10 @@ public string Fields()
                 {
                     // embed as an anonymous struct.  note that the ordering of this clause is
                     // important, i.e. we don't want to flatten primary types like dictionaries.
-                    indented.AppendFormat("*{0} {1}\n", property.ModelType.Name, property.JsonTag());
+                    indented.AppendFormat((property.ModelType as CompositeTypeGo).IsPolymorphic ?
```
Comment (jhendrixMSFT) [inline src/Model/CompositeTypeGo.cs]: Can you please update the comment explaining why we might need to emit as a pointer.

File: src/Templates/ModelTemplate.cshtml [hunk 2]
```diff
@@ -156,7 +157,14 @@ func (@(Model.Name.FixedValue.ToShortName()) *@(Model.Name)) UnmarshalJSON(body
         }
             @if ((p.ModelType is CompositeType && (p.ModelType as CompositeTypeGo).IsPolymorphic) || (p.ModelType is EnumType))
             {
+                @if (p.ShouldBeFlattened())
+                {
+        @:@(Model.Name.FixedValue.ToShortName()).@(p.ModelType.Name) = @(CodeNamerGo.Instance.GetVariableName(p.SerializedName))
```
Comment (jhendrixMSFT) [inline src/Templates/ModelTemplate.cshtml]: It would be good to refactor this to avoid the duplication.

File: src/Model/CompositeTypeGo.cs (hunk 1 above)
Comment (mcardosos) [reply src/Model/CompositeTypeGo.cs]: Fixed

File: src/Templates/ModelTemplate.cshtml (hunk 2 above)
Comment (mcardosos) [reply src/Templates/ModelTemplate.cshtml]: Fixed

**Comment by mcardosos** [comment]
restart linux-x64

**Comment by mcardosos** [comment]
win32-x64 restart

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Pull request merged**